testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/HttpClientTest.groovy

```diff
@@ -137,6 +138,31 @@ abstract class HttpClientTest extends InstrumentationSpecification {
     method << BODY_METHODS
   }
 
+  def "basic #method request with CLIENT parent"() {
+    given:
+    assumeTrue(testWithClientParent())
+
+    when:
+    def status = runUnderParentClientSpan {
+      doRequest(method, server.address.resolve("/success"))
+    }
+
+    then:
+    status == 200
+    assertTraces(2) {
+      trace(0, 1) {
+        basicSpan(it, 0, "parent-client-span")
+      }
+      trace(1, 1) {
+        serverSpan(it, 0)
+      }
+    }
```

Comment:
I think this deserves more comments. This claims that trace propagation should NOT happen in this test, which is very different from all other tests and thus will be confusing for readers.